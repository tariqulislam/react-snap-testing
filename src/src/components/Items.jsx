import React from 'react'
import PropTypes from 'prop-types'

function Items (props) {
    const {items = []} = props;

    if (!items.length) {
      return <span> No Items In List </span>;
    }

    if(items.length === 1) {
        return <span>{items[0]}</span>;
    }

    return (
        <ul>
          {items.map(item => <li key={item}>{item}</li>)}
        </ul>
    )
}

Items.propTypes = {
    items: PropTypes.array,
}

export default Items;