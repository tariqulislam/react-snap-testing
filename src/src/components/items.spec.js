import React from 'react'
import renderer from 'react-test-renderer'

import Items from './Items'

it('renderers correctly when there is no items', () => {
    const tree = renderer.create(<Items />).toJSON();
    expect(tree).toMatchSnapshot();
})

it('renders correctly when there is a item in Item', () => {
    const items = ["tree", "man", "woman"];
    const tree = renderer.create(<Items items={items} />).toJSON()
    expect(tree).toMatchSnapshot()
})