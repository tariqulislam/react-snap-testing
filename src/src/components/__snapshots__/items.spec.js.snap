// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderers correctly when there is no items 1`] = `
<span>
   No Items In List 
</span>
`;

exports[`renders correctly when there is a item in Item 1`] = `
<ul>
  <li>
    tree
  </li>
  <li>
    man
  </li>
  <li>
    woman
  </li>
</ul>
`;
